src/main/java/org/mule/module/apikit/api/uri/URIResolver.java

```diff
@@ -226,12 +226,23 @@ private URIPattern findBest(Set<URIPattern> patterns) {
     URIPattern best = null;
     for (URIPattern p : patterns) {
       if (p.match(this._uri)) {
-        if (best == null || p.score() > best.score()) {
-          best = p;
+        if (hasTrailingForwardSlash(p)) {
+          if (best == null || p.score() > best.score()) {
+            best = p;
+          }
         }
       }
     }
     return best;
   }
 
+  private boolean hasTrailingForwardSlash(URIPattern p) {
+
+    if (this._uri.endsWith("/")) {
+      return p.toString().endsWith("/");
+    } else {
+      return true;
+    }
```

Comment:
For cases when there is no trailing slashes in the URI, we want to return true

Lets say these are the available patterns:
test/{xyz}
test/


If incoming URI is `test/` , `p.match(this._uri)` returns true for both patterns. It should only be matched to `test/` pattern. Hence, the `if (hasTrailingForwardSlash(p))` decision statement

 If the incoming URI is `test/abctest` , it will only be matched to `test/{xyz}` pattern

Reply:
Thanks. This clarifies a bit - but i think the condition we check is still ambiguous. What if there is one more pattern `test/{xyz}/` .

Reply:
For `test/{xyz}/` , this won't match `test/` with the matcher condition (`p.match(this._uri)` condition) because elements between slashes are considered separate resources. eg: `test/123/`

Path parameter such as `{xyz}` is expected to be non empty, as that will allow  badly formed URLs such as
`test// ` to be matched

So test/{xyz} should only match with URL such as test/123 , and `test/` should instead always have its own pattern.

Additional some what discussion on the q-a channel https://salesforce-internal.slack.com/archives/C06REGCRJEQ/p1718129523541329

Reply:
Thanks. This discussion was very useful.